services/QuillLMS/lib/tasks/activities.rake

```diff
@@ -44,4 +44,27 @@ namespace :activities do
       puts "Failed to update for activity named '#{row['name']}'"
     end
   end
+
+  desc 'Take a pipe of CSV with activity IDs and new names, set the specified Activities to their new name'
+  task bulk_update_names: :environment do
+    pipe_data = $stdin.read unless $stdin.tty?
+
+    unless pipe_data
+      puts 'No data detected on STDIN.  You must pass data to the task for it to run.  Example:'
+      puts '  rake activities:bulk_update_names < path/to/local/file.csv'
+      puts ''
+      puts 'If you are piping data into Heroku, you need to include the --no-tty flag:'
+      puts '  heroku run rake activities:bulk_update_names -a empirical-grammar --no-tty < path/to/local/file.csv'
+      exit 1
+    end
+
+    CSV.parse(pipe_data, headers: true) do |row|
+      activity = Activity.find(row['Activity ID'])
+      activity.update!(name: row['New Name'])
```

Comment:
In the past, I've found that CSVs can contain unexpected values. `Activity.name` is not a no null column. Also, values like `\t` or `\n` will cause unpleasant behavior. So I recommend some validation here.

Reply:
Done.